### PR TITLE
5 as client i want to get a wallets cryptocurrencies

### PR DESCRIPTION
**Pull request by martinez2001**
Get Wallet cryptocurrencies issue implemented. Look at my code in order to find code smells.

### CODE REVIEW

**Review by armenllanos — COMMENTED** [review]
Easy to make the change, just erase the conflicting line of the laravel.yml

File: app/Infrastructure/Controllers/GetWalletCryptocurrenciesController.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    public function __invoke(string $wallet_id): JsonResponse
+    {
+        try {
+            $coins = $this->getWalletCryptoCurrenciesService->execute($wallet_id);
+        }catch (Exception $exception) {
+            if ($exception->getMessage() === 'Wallet not found')
+                return response()->json([
+                    "error" => "a wallet with the specified ID was not found"
+                ], Response::HTTP_NOT_FOUND);
+
```
Comment (rubeniriso) [inline app/Infrastructure/Controllers/GetWalletCryptocurrenciesController.php:35]: \n

File: app/Infrastructure/Controllers/GetWalletCryptocurrenciesController.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            else if ($exception->getMessage() === 'Service Unavailable')
+                return response()->json([
+                    "error" => "service is unavailable"
+                ], Response::HTTP_SERVICE_UNAVAILABLE);
+        }
+        return response()->json([
+            "coins_array" => json_encode($coins)
+        ],Response::HTTP_OK);
+
```
Comment (rubeniriso) [inline app/Infrastructure/Controllers/GetWalletCryptocurrenciesController.php:44]: \n

File: app/Infrastructure/Controllers/GetWalletCryptocurrenciesController.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    public function __construct(GetWalletCryptoCurrenciesService $getWalletCryptoCurrenciesService)
+    {
+        $this->getWalletCryptoCurrenciesService = $getWalletCryptoCurrenciesService;
+    }
+
+    public function __invoke(string $wallet_id): JsonResponse
+    {
+        try {
+            $coins = $this->getWalletCryptoCurrenciesService->execute($wallet_id);
+        }catch (Exception $exception) {
```
Comment (rubeniriso) [inline app/Infrastructure/Controllers/GetWalletCryptocurrenciesController.php:30]: space between } and catch

File: tests/app/Application/GetWalletCryptocurrencies/GetWalletCryptocurrenciesServiceTest.php [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            ->expects('getWallet')
+            ->with($wallet_id)
+            ->once()
+            ->andReturn(null);
+
+        $response = $this->getWalletCryptocurrenciesService->execute($wallet_id);
+
+        $this->assertEquals(null, $response);
+    }
+
```
Comment (rubeniriso) [inline tests/app/Application/GetWalletCryptocurrencies/GetWalletCryptocurrenciesServiceTest.php:42]: \n

File: tests/app/Infrastructure/Controller/GetWalletCryptoCurrenciesControllerTest.php [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+            ->expects("getWallet")
+            ->with('1414')
+            ->once()
+            ->andThrow(new Exception('Service Unavailable'));
+
+        $response = $this->get('api/wallet/1414');
+        $response->assertStatus(Response::HTTP_SERVICE_UNAVAILABLE)
+            ->assertExactJson(['error' => 'service is unavailable']);
+    }
+
```
Comment (rubeniriso) [inline tests/app/Infrastructure/Controller/GetWalletCryptoCurrenciesControllerTest.php:44]: \n

**Review by rubeniriso — COMMENTED** [review]
A bit of formatting, rest works,
good job.

**Pull request closed without merging**